### ISSUE TITLE
MediaPipe Holistic: what are possible reasons for not getting any landmarks?

### ISSUE THREAD

**Issue by m-decoster** [google/mediapipe#3767]
**TL;DR:** What are the reasons that `results.left_hand_landmarks` (or similar fields) can be `None`?

I am extracting hand landmarks using code similar to this:

```
def extract_keypoints(results):
    lh = np.array([[res.x, res.y, res.z] for res in results.left_hand_landmarks.landmark]).flatten() if results.left_hand_landmarks else np.zeros(21*3)
    rh = np.array([[res.x, res.y, res.z] for res in results.right_hand_landmarks.landmark]).flatten() if results.right_hand_landmarks else np.zeros(21*3)
    return np.concatenate([ lh, rh])
```

For some videos, I get hand landmarks in one frame, but none in the next, even though the frames are (visually) virtually identical.
Where can I find information in the documentation or source code that tells me more about *why* `results.left_hand_landmarks` can be `None`? I have tried with several thresholds but there is no setting that always results in predictions as far as I can tell.

In some cases the wrist position is correct, so the hand ROI detector should be able to get a decent crop and thus I would expect keypoints to be detected. But even in such cases, sometimes I get no results from MediaPipe.

**Comment by m-decoster** [comment]
@kuaashish [Here](https://users.ugent.be/~mcdcoste/shared/mediapipe_fail_example.pdf) is an example of a video, where two subsequent frames are very similar, but there are no predictions for one frame and there are predictions for the other frame.

The other failure cases are very similar.

The blue box was added *after* MediaPipe extraction for privacy reasons, so it does not influence the mediapipe extraction.

**Comment by jdambre** [comment]
Still waiting for an answer to this question (https://github.com/google/mediapipe/issues/3339) ...

**Comment by jinfagang** [comment]
![image](https://user-images.githubusercontent.com/21303438/216224227-2aa47925-2069-474d-bc65-2c5fccc7f16a.png)

same issue here, the pose is not working

**Comment by jdambre** [comment]
Could someone please respond to this? Issues like these, and the fact that we get NO information about where they originate, are causing a lot of frustration. Current hand keypoint detection is simply too unstable and unreliable for many applications. This is no problem for things like gesture control in games, where an undetected gesture can simply be repeated (although users may get frustrated). It is a problem in applications where you really need continuous reliability, e.g. when monitoring someone's hand movements during procedures (medical, operational) for learner feedback or for tasks like sign language translation from video. 

On the one hand we are very grateful for this tool, but on the other hand many researchers are constrained by the unpredictability and lack of transparency, up to the point where, at the end of the day, we still have no other option than developing our own keypoint extraction models for the hands! 

Given the expertise that exists in the research field, it would be more useful to allow us to collaborate/contribute to improvements that benefit all of us than continuing like this with untransparency and continuous parallell efforts for trying to solve the same basic problems in the tools we need.

**Comment by jinfagang** [comment]
I solved my problem, thank u everyone.

**Comment by jdambre** [comment]
> I solved my problem, thank u everyone.
@jinfagang could you share the solution? or was it simply a bug ...

**Comment by kuaashish** [comment]
Hello @m-decoster,
We are upgrading the [MediaPipe Legacy Solutions](https://developers.google.com/mediapipe/solutions/guide#legacy) to [new MediaPipe solutions](https://developers.google.com/mediapipe/solutions) However, the libraries, documentation, and source code for all the MediapPipe Legacy Solutions will continue to be available in our [GitHub repository](https://github.com/google/mediapipe) and through library distribution services, such as [Maven](https://maven.google.com/web/index.html?#com.google.mediapipe) and [NPM](https://www.npmjs.com/search?q=%40mediapipe).

You can continue to use those legacy solutions in your applications if you choose. Though, we would request you to check [new MediaPipe solutions](https://developers.google.com/mediapipe/solutions) which can help you more easily build and customize ML solutions for your applications. These new solutions will provide a superset of capabilities available in the legacy solutions. Thank you

**Comment by jdambre** [comment]
Hi @kuaashish ,

This reads like a generic answer that has very little to do with our questions, on which we haven't received ANY replies yet!

You point to new versions, but running MP again on all data takes a very long time, so this is only useful if the detection has improved. Can you tell us whether anything has changed to the (quality of) the core keypoint extraction algorithms?

**Comment by m-decoster** [comment]
Hello @kuaashish 

I have noticed the new solutions, but pose estimation is not yet included, so this is not useful for us right now.
Are there any improvements to the underlying model that would improve its robustness in these new solutions?

Are there any indications as to why the robustness is lacking in the situation that I detailed in [this comment](https://github.com/google/mediapipe/issues/3767#issuecomment-1291861360)?

**Comment by kuaashish** [comment]
@m-decoster,
Pose solution is now available(Check [here](https://developers.google.com/mediapipe/solutions/examples)). All upgraded solutions are superset of the old legacy solution, you will able see the better results, Improvement in the performance. Will suggest you to try our upgraded solutions.Further, please feel free to raise a new issue if same behaviour in the upgraded solutions. Thank you!
